src/GraphMsg.chpl

```diff
@@ -0,0 +1,4510 @@
+                               //forall i in UnVisitedSet  with (ref SetNextF) {
+                                  if depth[i]==-1 {
+                                      var    numNF=nf[i];
+                                      var    edgeId=sf[i];
+                                      var nextStart=max(edgeId,edgeBegin);
+                                      var nextEnd=min(edgeEnd,edgeId+numNF-1);
+                                      ref NF=df[nextStart..nextEnd];
+                                      forall j in NF with (ref SetNextF){
+                                             if (SetCurF.contains(j)) {
+                                                   depth[i]=cur_level+1;
+                                                   SetNextF.add(i);
+                                             }
+                                      }
+
+                                  }
+                               }
+                               //UnVisitedSet.clear();
+                               //forall i in vertexBeginR..vertexEndR with (ref UnVisitedSet) {
+                               //   if depth[i]==-1 {
+                               //      UnVisitedSet.add(i);
+                               //   }
+                               //}
+                               forall i in vertexBeginR..vertexEndR  with (ref SetNextF) {
+                               //forall i in UnVisitedSet  with (ref SetNextF) {
+                                  if depth[i]==-1 {
+                                      var    numNF=nfR[i];
+                                      var    edgeId=sfR[i];
+                                      var nextStart=max(edgeId,edgeBegin);
+                                      var nextEnd=min(edgeEnd,edgeId+numNF-1);
+                                      ref NF=dfR[nextStart..nextEnd];
+                                      forall j in NF with (ref SetNextF)  {
+                                             if (SetCurF.contains(j)) {
+                                                   depth[i]=cur_level+1;
+                                                   SetNextF.add(i);
+                                             }
+                                      }
+                                  }
+                               }
+                           }
+                       }//end on loc
+                    }//end coforall loc
+                    cur_level+=1;
+                    //numCurF=SetNextF.getSize();
+                    numCurF=SetNextF.size;
+
+                    if (numCurF>0) {
+                        var tmpary:[0..numCurF-1] int;
+                        var sortary:[0..numCurF-1] int;
+                        var numary:[0..numCurF-1] int;
+                        var tmpa=0:int;
+                        forall (a,b)  in zip (tmpary,SetNextF.toArray()) {
+                            a=b;
+                        }
```

Comment:
Two issues here:
1. From the documentation, there is no order guaranteed to `set.toArray()` so it's going to be completely arbitrary, I'm not sure if that matters or not.
2. From the documentation, `set.toArray()` returns a copy of the values as an array, so looping through it is redundant, 
you should be able to simplify as `var tmpary = SetNextF.toArray();`

Reply:
Thanks and I will update the code based on your suggestions.